src/constants.h

```diff
@@ -43,6 +43,10 @@ constexpr static const uint32_t OPTION_ERROR = 3;
 constexpr static const uint32_t GENERIC_ERROR = 99;
 constexpr static const size_t DEFAULT_MAX_THREADS = 16;
 
+/// Size of the data being read from stream (e.g. pipe) when reading input data
+/// from user-provided data stream. Defaults to 4 bytes.
+constexpr uint32_t DEFAULT_STREAM_DATA_SIZE = 4;
```

Comment:
I don't like how this piece of information needs to be known in two distinct locations in the code, but I can't think of a clean way to encapsulate it...

Perhaps this will change anyway if DataLoader needs to be modified to provide more genuine/piecewise piping.

Reply:
Since I couldn't come up with a better way to encapsulate this, I'll just resolve the conversation.